modules/PhenoDigm.py

```diff
@@ -0,0 +1,362 @@
+            self.load_tsv(HGNC_DATASET_FILENAME)
+            .withColumnRenamed('hgnc_id', 'hgnc_gene_id')
+            .withColumnRenamed('ensembl_gene_id', 'targetFromSourceId')  # Using the final name.
+            .select('hgnc_gene_id', 'targetFromSourceId')
+        )
+        self.mgi_gene_id_to_ensembl_mouse_gene_id = (  # E.g. 'MGI:87853', 'ENSMUSG00000027596'.
+            self.load_tsv(MGI_DATASET_FILENAME)
+            .withColumnRenamed('1. MGI accession id', 'mgi_gene_id')
+            .withColumnRenamed('11. Ensembl gene id', 'targetInModel')  # Using the final name.
+            .filter(pf.col('targetInModel').isNotNull())
+            .select('mgi_gene_id', 'targetInModel')
+        )
+
+        # Mouse to human gene mappings, e.g. 'MGI:1346074', 'HGNC:4024'.
+        self.mouse_gene_to_human_gene = self.load_solr_csv('gene_gene')
+        # Mouse to human phenotype mappings, e.g. 'MP:0000745','HP:0100033'.
+        self.mouse_phenotype_to_human_phenotype = self.load_solr_csv('ontology_ontology')
+
+        # Mouse model and disease data.
+        # Note that the models are accessioned with the same prefix ('MGI:') as genes, but they are separate entities.
+        self.mouse_model = self.load_solr_csv('mouse_model')  # E. g. 'MGI:3800884', ['MP:0001304 cataract'].
+        self.disease = self.load_solr_csv('disease')  # E.g. 'OMIM:609258', ['HP:0000545 Myopia'].
+        # E. g. 'MGI:2681494', 'C57BL/6JY-smk', 'smk/smk', 'ORPHA:3097', 'Meacham Syndrome', 91.6, 'MGI:98324'.
+        self.disease_model_summary = self.load_solr_csv('disease_model_summary')
+        self.ontology = (
+            self.load_solr_csv('ontology')  # E.g. 'HP', 'HP:0000002', 'Abnormality of body height'.
+            .filter((pf.col('ontology') == 'MP') | (pf.col('ontology') == 'HP'))
+        )
+        assert self.ontology.select('phenotype_id').distinct().count() == self.ontology.count(), \
+            f'Encountered multiple names for the same term in the ontology table.'
+
+    def generate_phenodigm_evidence_strings(self, score_cutoff):
+        """Generate the evidence by renaming, transforming and joining the columns."""
+        # Process ontology information to enable MP and HP term lookup based on the ID.
+        mp_terms, hp_terms = (
+            self.ontology
+            .filter(pf.col('ontology') == ontology_name)
+            .withColumnRenamed('phenotype_id', f'{ontology_name.lower()}_id')
+            .withColumnRenamed('phenotype_term', f'{ontology_name.lower()}_term')
+            .select(f'{ontology_name.lower()}_id', f'{ontology_name.lower()}_term')
+            for ontology_name in ('MP', 'HP')
+        )
+
+        # Split lists of phenotypes in the `mouse_model` and `disease` tables and keep only the ID. For example, one row
+        # with 'MP:0001529 abnormal vocalization,MP:0002981 increased liver weight' becomes two rows with 'MP:0001529'
+        # and 'MP:0002981'.
+        model_mouse_phenotypes = (
```

Comment:
Thank you for a great suggestion, addressed in https://github.com/opentargets/evidence_datasource_parsers/pull/80/commits/a33c14012419d72a56140dcce863566741f822c6

Reply:
In the end I realised that `regexp_extract` doesn't allow the things to be sufficiently simplified, because it can only extract one match. I was able to solve it with upgrading to Spark 3.1.1 and using `expr(regexp_extract_all)`: https://github.com/opentargets/evidence_datasource_parsers/pull/80/commits/015cb611db024e84a61968ec167321fe8342e5d9. What do you think?